src/SyntaxHighlighter/Cpp.cpp

```diff
@@ -53,35 +56,47 @@ const char* const kStringRegex = "\"([^\\\\\"]|\\\\.)*\"|\'[^\']*\'";
 const char* const kOpenStringRegex = "\"([^\\\\\"]|\\\\.)*\\\\$";
 const char* const kEndStringRegex = "([^\\\\\"]|\\\\.)*\"";
 const char* const kNoEndStringRegex = "([^\\\\\"]|\\\\.)*\\\\$";
-const char* const kCapitalizedRegex = "(?<=[\\s\\(<])[A-Z][\\w\\*\\&]*";
+const char* const kCapitalizedRegex = "(?<=[\\s\\(<])[A-Z][\\w]*";
 const char* const kNoLowercaseRegex = "(?<=[\\s\\(])[A-Z][0-9A-Z\\_\\*\\&]*(?=[\\;\\,\\(\\s\\)])";
+const char* const kCommaRegex = "[\\;\\,]";
+// All related with namespaces and classes (e.g Namespace:: , std::cout, absl::Milliseconds) except
+// function definitions
+const char* const kNamespaceRegex = "[A-Z]\\w*::|[a-z]\\w*::\\w*(?=\\()";
```

Comment:
Why is there a differentiation between upper and lower case?

Reply:
Nice catch! I started making different things and after changed it and never review after that.

Reply:
I should rewrite this to include lower and uppercase together. Thanks for pointing out. It's something that I used to need, but not anymore, I think.

It isn't a difference between lower and upper, it's about the left and right. The first part is catching the left part of the " : : " and the second the right part which shouldn't be a function (because we are painting them yellow).

Reply:
Divided into two:
kNamespaceRegex match the first part before "::" and kNamespaceVariableRegex match the right part. Turn out to be simpler after that division and cleaner.